pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -464,7 +464,7 @@ stdenv.mkDerivation ({
   installPhase = ''
     runHook preInstall
 
-    ${if !isLibrary then "${setupCommand} install" else ''
+    ${if !isLibrary then "${setupCommand} copy ${buildTarget}" else ''
```

Comment:
Do `install` and `copy` do the exact same thing here?  Is there no difference in the output derivation?

Would you be able to provide some proof that this doesn't actually break anything?

I'm imagining taking a bunch of popular Haskell packages (where `isLibrary` is set to false), compiling them both with this PR and without, and showing that the output derivations are identical.

I'm somewhat surprised that this would cause so many rebuilds, especially since it should only affect non-library Haskell executables.  I thought we just didn't have too many of those in nixpkgs.

Reply:
> Do install and copy do the exact same thing here? Is there no difference in the output derivation?

If there is a difference, I didn't find it.

> Would you be able to provide some proof that this doesn't actually break anything?

I did this:

```bash
nix-build ./. -A haskellPackages.pandoc
./result/bin/pandoc --help
```

and it worked. Not sure what more to do here.

> I'm imagining taking a bunch of popular Haskell packages (where isLibrary is set to false), compiling them both with this PR and without, and showing that the output derivations are identical.

With the new version I pushed, these libraries should be unaffected. In fact, I think that there are no packages currently that use the `buildTarget` parameter and have `isLibrary == false` because they would not build today anyway.